File: apollo/src/main/java/io/muun/apollo/domain/errors/MessageOriginError.java
package io.muun.apollo.domain.errors;

import io.muun.common.api.messages.MessageOrigin;
import io.muun.common.api.messages.MessageSpec;

public class MessageOriginError extends RuntimeException {

    /**
     * Rich constructor.
     */
    public MessageOriginError(String sessionUuid,
                              long messageId,
                              MessageOrigin origin,
                              MessageSpec spec) {

        super(String.format(
                "Sender %s sent message %s of type %s from %s, but we expected %s",
                sessionUuid,
                Long.toString(messageId),
                spec.messageType,
                origin.name(),
                spec.allowedOrigin.name()
        ));
    }
}

File: android/apollo/src/main/java/io/muun/apollo/domain/errors/MessageOriginError.kt
package io.muun.apollo.domain.errors

import io.muun.common.api.messages.MessageOrigin
import io.muun.common.api.messages.MessageSpec

class MessageOriginError(sessionId: String, msgId: Long, origin: MessageOrigin, spec: MessageSpec):
    MuunError("Received a message from an unexpected origin") {

    init {
        metadata["sessionUuid"] = sessionId
        metadata["messageId"] = msgId
        metadata["messageType"] = spec.messageType
        metadata["origin"] = origin.name
        metadata["allowedOrigin"] = spec.allowedOrigin.name
    }

}
